ui/components/component-library/avatar-token/avatar-token.js

```diff
@@ -0,0 +1,96 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { BaseAvatar } from '../base-avatar';
+
+import {
+  COLORS,
+  SIZES,
+  DISPLAY,
+  ALIGN_ITEMS,
+  JUSTIFY_CONTENT,
+} from '../../../helpers/constants/design-system';
+
+export const AvatarToken = ({
+  size = SIZES.MD,
+  backgroundColor = COLORS.BACKGROUND_ALTERNATIVE,
+  borderColor = COLORS.BORDER_DEFAULT,
+  tokenName,
+  tokenImageUrl,
+  showHalo,
+  ...props
+}) => {
+  const [showFallback, setShowFallback] = useState(false);
+
+  useEffect(() => {
+    setShowFallback(!tokenImageUrl);
+  }, [tokenImageUrl]);
+
+  const baseStyle = showHalo ? { position: `relative` } : {};
+  const sizePercentage = showHalo ? '62.5%' : '100%';
+
+  const style = size
+    ? {
+        height: sizePercentage,
+        width: sizePercentage,
+        borderRadius: '50%',
+        zIndex: '2',
+      }
+    : {};
+
+  const handleOnError = () => {
+    setShowFallback(true);
+  };
+
+  return (
+    <BaseAvatar
+      size={size}
+      style={baseStyle}
+      display={DISPLAY.FLEX}
+      alignItems={ALIGN_ITEMS.CENTER}
+      justifyContent={JUSTIFY_CONTENT.CENTER}
+      data-test-id="avatar-token"
```

Comment:
We shouldn't be adding data-test-ids to the component itself because there may be multiple data-test ids on a page. It should be passed to each component when it is used.
```suggestion
```